src/main/java/co/cask/hydrator/plugin/BatchDynamoDBSink.java

```diff
@@ -0,0 +1,234 @@
+import co.cask.hydrator.common.ReferenceBatchSink;
+import co.cask.hydrator.common.ReferencePluginConfig;
+import co.cask.hydrator.plugin.utils.DynamoDBConstants;
+import com.amazonaws.services.dynamodbv2.document.Item;
+import com.amazonaws.services.dynamodbv2.document.PrimaryKey;
+import com.google.common.base.Splitter;
+import com.google.common.base.Strings;
+import org.apache.hadoop.io.NullWritable;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+
+/**
+ * A {@link BatchSink} that writes data to DynamoDB.
+ * This {@link BatchDynamoDBSink} takes a {@link StructuredRecord} in, converts it to Item, and writes it to the
+ * DynamoDB table.
+ */
+@Plugin(type = BatchSink.PLUGIN_TYPE)
+@Name(BatchDynamoDBSink.PLUGIN_NAME)
+@Description("CDAP Amazon DynamoDB Batch Sink takes the structured record from the input source and writes the data " +
+  "into Amazon DynamoDB.")
+public class BatchDynamoDBSink extends ReferenceBatchSink<StructuredRecord, NullWritable, Item> {
+  public static final String PLUGIN_NAME = "DynamoDB";
+  private final DynamoDBConfig config;
+
+  public BatchDynamoDBSink(DynamoDBConfig config) {
+    super(config);
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    config.validateTableName();
+    config.validatePrimaryKey();
+  }
+
+  @Override
+  public void prepareRun(BatchSinkContext context) {
+    context.addOutput(Output.of(config.referenceName, new DynamoDBOutputFormatProvider(config)));
+  }
+
+  @Override
+  public void transform(StructuredRecord input, Emitter<KeyValue<NullWritable, Item>> emitter) throws Exception {
+    Set<String> primaryKeyAttributes = Splitter.on(',').trimResults().withKeyValueSeparator(":")
+      .split(config.primaryKeyFields).keySet();
+    PrimaryKey primaryKey = new PrimaryKey();
+    Item item = new Item();
+    for (Schema.Field field : input.getSchema().getFields()) {
+      String fieldName = field.getName();
+      if (primaryKeyAttributes.contains(field.getName())) {
+        primaryKey.addComponent(fieldName, input.get(fieldName));
+      } else {
+        item.with(fieldName, input.get(fieldName));
+      }
+    }
+    item.withPrimaryKey(primaryKey);
+    emitter.emit(new KeyValue<>(NullWritable.get(), item));
+  }
+
+  /**
+   * Config class for Batch DynamoDB Sink
+   */
+  public static class DynamoDBConfig extends ReferencePluginConfig {
+    @Description("The hostname and port for AWS DynamoDB instance to connect to, separated by a colon. If not " +
+      "provided, it will be constructed using regionId. Eg. localhost:8000.")
+    @Nullable
+    private String endpointUrl;
+
+    @Nullable
+    @Description("The region for AWS Dynamo DB to connect to. Default is us-west-2 i.e. US West (Oregon).")
+    private String regionId;
```

Comment:
We had kept this assuming cases where the user is running DynamoDB in local mode for testing purpose. In this case, providing only the option for region id may not be enough. Also, providing a drop-down of options will not feasible for such cases as user can have the endpointURl as localhost:<anyAvailablePort>.
The current implementation allows the user to skip regionId or endpointURL or both. The plugin will create the regionId or endpointURL based on the other parameters provided(accessKey, and secretAccessKey).
Please, let me know if we still need to remove the endpointURL or take it as a drop-down widget instead of a text-box.